cfme/tests/cloud_infra_common/test_vm_instance_analysis.py

```diff
@@ -321,6 +328,63 @@ def schedule_ssa(appliance, ssa_vm, ssa_profile, wait_for_task_result=True):
     return ss
 
 
+@pytest.fixture
+def compare_linux_vm_data(soft_assert):
+
+    def _compare_linux_vm_data(ssa_vm):
+        expected_users = ssa_vm.ssh.run_command("cat /etc/passwd | wc -l").output.strip('\n')
+        expected_groups = ssa_vm.ssh.run_command("cat /etc/group | wc -l").output.strip('\n')
+        expected_packages = ssa_vm.ssh.run_command(
+            ssa_vm.system_type['package-number']).output.strip('\n')
+        expected_services = ssa_vm.ssh.run_command(
+            ssa_vm.system_type['services-number']).output.strip('\n')
+
+        view = navigate_to(ssa_vm, 'Details')
+        current_users = view.entities.summary('Security').get_text_of('Users')
+        current_groups = view.entities.summary('Security').get_text_of('Groups')
+        current_packages = view.entities.summary('Configuration').get_text_of('Packages')
+        current_services = view.entities.summary('Configuration').get_text_of('Init Processes')
+
+        soft_assert(current_users == expected_users,
+                    "users: '{}' != '{}'".format(current_users, expected_users))
+        soft_assert(current_groups == expected_groups,
+                    "groups: '{}' != '{}'".format(current_groups, expected_groups))
+        soft_assert(current_packages == expected_packages,
+                    "packages: '{}' != '{}'".format(current_packages, expected_packages))
+        soft_assert(current_services == expected_services,
+                    "services: '{}' != '{}'".format(current_services, expected_services))
+
+    return _compare_linux_vm_data
```

Comment:
This looks cleaner now :) I am just not sure what the purpose of putting this in a nested method is. It appears this would always return 'None' -- because `_compare_linux_vm_data` returns nothing. Do you have to do the 'return' because it is a fixture? If that's the case, couldn't you remove the `@pytest.fixture` and simply pass `soft_assert` as an arg when calling this method in the test?

Reply:
In this case this fixture return a function that we can use in our test, and I would like to keep it as a fixture and pass only vm object, and not confuse also with passing soft_assert, as we are already can use it in the fixture

Reply:
Gotcha, I overlooked that it returns the whole function... ok